configs/components/ruby-shadow.rb

```diff
@@ -41,6 +41,15 @@
     end
   end
 
+  matchdata = platform.settings[:ruby_version].match /(\d+)\.(\d+)\.\d+/
+  ruby_major_version = matchdata[1].to_i
+  if ruby_major_version >= 3
+    base = "resources/patches/ruby_32"
+    # https://github.com/apalmblad/ruby-shadow/issues/26
+    # if ruby-shadow gets a 3 release this should be removed
+    pkg.apply_patch "#{base}/ruby-shadow-2.5.0-cflags.patch", strip: "0"
```

Comment:
This part was tricky, usually when we apply_patch it defaults to strip: "1", which is non-relative pathing, but in this case vanagon has change directory into the ruby-shadow directory and we want to apply relative, so strip needs to be set to 0, and apply_patch takes an option hash to allow us to set that.